include/lemlib/util.hpp

```diff
@@ -124,4 +124,17 @@ Waypoint closestWaypoint(const std::vector<Waypoint>& waypoints, const Pose& tar
  * multiple intersections, the first one will be returned
  */
 Pose circleLineIntersect(Pose p1, Pose p2, Pose center, float radius);
+
+/*
+ * @brief Linearly interpolate two points
+ *
+ * @param x the x value to interpolate
+ * @param x1 the x value of the lower adjacent coordinate
+ * @param y1 the y value of the lower adjacent coordinate
+ * @param x2 the x value of the upper adjacent coordinate
+ * @param y2 the y value of the upper adjacent coordinate
+ * @return the interpolated y value
+ */
+float linearInterp(float x, float x1, float y1, float x2, float y2);
```

Comment:
have you considered using Pose::lerp?

Reply:
I considered using Pose::lerp, but I decided against it because the X and Y values here don't represent X and Y coordinates, just the domain and co-domain of the interpolation function. For example in `interpolateLinear` the "Y" values are the gain constants, so it wouldn't make sense to use Pose::lerp because it isn't really a position. Please let me know if you would like me to change this though.